sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -117,9 +122,11 @@ class SparkSession private(
   @InterfaceStability.Unstable
   @transient
   lazy val sessionState: SessionState = {
-    SparkSession.reflect[SessionState, SparkSession](
-      SparkSession.sessionStateClassName(sparkContext.conf),
-      self)
+    parentSessionState
+      .map(_.copy(this))
```

Comment:
Good point. I think we can force materializing these lazy val before cloneSession returns. Unfortunately, we cannot remove lazy because of the complicated cycle dependencies in these objects.